app/assets/javascripts/repository_columns/index.js

```diff
@@ -0,0 +1,247 @@
+        HelperModule.flashAlertMsg(
+          I18n.t('libraries.repository_columns.no_permissions'), 'danger'
+        );
+      });
+    });
+  }
+
+  function removeElementFromDom(column) {
+    $('.list-group-item[data-id="' + column.id + '"]').remove();
+    if ($('.list-group-item').length === 0) {
+      location.reload();
+    }
+  }
+
+  function initDeleteSubmitAction(modal, form) {
+    modal.find('[data-action="delete"]').on('click', function() {
+      form.submit();
+      modal.modal('hide');
+      animateSpinner();
+      form.on('ajax:success', function(e, data) {
+        removeElementFromDom(data);
+        HelperModule.flashAlertMsg(data.message, 'success');
+        animateSpinner(null, false);
+      }).on('ajax:error', function(e, xhr) {
+        animateSpinner(null, false);
+        HelperModule.flashAlertMsg(xhr.responseJSON.message, 'danger');
+      });
+    });
+  }
+
+  function initDeleteColumnModal() {
+    $('[data-action="destroy"]').off('click').on('click', function() {
+      var element = $(this);
+      var modalHtml = $('#deleteRepositoryColumn');
+      $.get(element.closest('li').attr('data-destroy-url'), function(data) {
+        modalHtml.find('.modal-body').html(data.html)
+          .promise()
+          .done(function() {
+            modalHtml.modal('show');
+            initDeleteSubmitAction(modalHtml, $(modalHtml.find('form')));
+          });
+      }).fail(function() {
+        HelperModule.flashAlertMsg(
+          I18n.t('libraries.repository_columns.no_permissions'), 'danger'
+        );
+      });
+    });
+  }
+
+  function insertNewListItem(column) {
+    var html = `<li class="list-group-item row" data-id="${column.id}"
+                    data-destroy-url="${column.destroy_html_url}"
+                    data-edit-url="${column.edit_url}">
+                  <div class="col-xs-8">
+                    <span class="pull-left column-name">${column.name}</span>
+                  </div>
+                  <div class="col-xs-4">
+                    <span class="controlls pull-right">
+                      <button class="btn btn-default" data-action="edit">
+                        <span class="fas fa-pencil-alt"></span>&nbsp;
+                        <%= I18n.t "libraries.repository_columns.index.edit_column" %>
+                      </button>&nbsp;
+                      <button class="btn btn-default delete-column" data-action="destroy">
+                        <span class="fas fa-trash-alt"></span>&nbsp;
+                        <%= I18n.t "libraries.repository_columns.index.delete_column" %>
+                      </button>
+                    </span>
+                  </div>
+                </li>`;
+
+    // remove element if already persent
+    $('[data-id="' + column.id + '"]').remove();
+    $(html).insertBefore('.repository-columns-body ul li:first')
+      .promise()
+      .done(function() {
+        initDeleteColumnModal();
+        initEditCoumnModal();
```

Comment:
Why we need every time init this. Make it as delegated actions on container itself.

Reply:
:+1: